_plugins/tabs.rb

```diff
@@ -31,7 +31,9 @@ def initialize(tag_name, text, tokens)
         def checkMandatories(items)
             # exception handeling if user not gave any id
             if items.key?("id") == false || items["id"].match(IdPattern) == false
-                raise "id is required and can only contain Numbers,- and Alphabet."
+                print("id not found generating a random id.")
+                num = rand(0..9999)
```

Comment:
This will result in random build failures in the event the same random number is chosen twice.  Is it possible to use an incrementing number instead?

Reply:
Since each time {% tab %} generates a new class, incremental approach can not be implemented without hacking into Jekyll.
I have changed random number to 8 random characters which can minimize the chance of duplication as low as possible.